docs/git/branches/branches.md

```diff
@@ -0,0 +1,231 @@
+
+## Learning objectives
+
+By the end of this class, you should be able to:
+
+- Explain why branches can be useful when using Git.
+- Create a branch, and add commits to it.
+- Switch between branches.
+- Create a pull request from a branch to the main branch.
+- Work with another trainee on non-conflicting changes to a single branch.
+
+## 1) What is a branch?
+
+We saw in the first Git lesson that Git stores the history of your changes as a list of commits, one after the other:
+
+![Committing to the Master Branch](./assets/git-main-commit.gif)
+
+Since then, you've been pressing a button in GitHub Desktop called "Commit to main", or running `git commit`, to make a new commit. But what does that "to main" mean? main is called a _branch_, which is a list of commits. It turns out, you can have more than one branch, and they can share some commits if they want:
+
+![Committing to another Branch](./assets/git-branch-commit.gif)
+
+### Why would we want to do this?
+
+There are a few reasons. The main one is: So you can work on more than one thing at the same time.
+
+Let's look at our repo of recipes. Right now, it contains one recipe, for a brownie in a mug. There are a couple of changes we want to make:
+
+1. We want to add an optional topping to put on top of the brownie (maybe ice cream, maybe you have a better idea). We'll need to add it both as an extra ingredient, and an extra step at the end.
+2. A lot of recipes start with a story of how the author found them, or some time they've really enjoyed the recipe. We want to add a story, before the "What you need" section.
+
+Let's imagine we start writing the story, but we want to quickly add the optional ice cream step. And let's imagine that we want someone else to review both changes before we merge them to main.
+
+We could make both changes to main, and ask someone to review them both, but then they both need to be right before we can merge either one - we can't merge just the ice cream step until the story is finished.
+
+Or we could only make the ice cream change, then send it for review, then when that merges, write the story and send it for review. But that slows us down, and also means that the reviewer can't review both changes at the same time.
+
+It would be better if we could send out one pull request for the ice cream change, and another for the story. We can do this - with branches:
+
+![Committing to another Branch](./assets/git-example-commit.gif)
+
+We can then add new commits on top of each branch, separately, and later on merge both branches to main.
+
+### How do we create a new branch?
+
+```console
+$ git switch --create story
```

Comment:
Good call with going with `switch` . I always felt having to create + checkout across two steps was a stumbling block for learners